encord/client.py

```diff
@@ -516,22 +516,25 @@ def add_private_data_to_dataset(
             },
         )["process_hash"]
 
-        for _ in range(10):
-            polling_response = self._querier.basic_getter(
-                DatasetDataLongPolling,
-                self._config.resource_id,
-                payload={
-                    "process_hash": process_hash,
-                    "ignore_errors": ignore_errors,
-                },
-            )
-
-            if polling_response["is_done"]:
+        while True:
+            polling_response = None
+
+            for _ in range(3):
+                try:
+                    polling_response = self._querier.basic_getter(
+                        DatasetDataLongPolling,
+                        self._config.resource_id,
+                        payload={
+                            "process_hash": process_hash,
+                            "ignore_errors": ignore_errors,
+                        },
+                    )
+                except Exception as e:
```

Comment:
that should only catch network errors.

Reply:
I think we wanted something more complex, fine for me:
`except requests.exceptions.RequestException as e:`